regression/cbmc-library/float-nan-check/test.desc

```diff
@@ -6,20 +6,21 @@ main.c
 \[main.NaN.3\] line \d+ NaN on / in \(float\)n / myinf: FAILURE
 \[main.NaN.4\] line \d+ NaN on \* in myinf \* myzero: FAILURE
 \[main.NaN.5\] line \d+ NaN on \+ in -myinf \+ myinf: FAILURE
-\[main.NaN.6\] line \d+ NaN on - in myinf - myinf: FAILURE
-\[main.NaN.7\] line \d+ NaN on - in -myinf - -myinf: FAILURE
-\[main.NaN.8\] line \d+ NaN on \+ in byte_extract_little_endian\(c, (0|0l|0ll), float\) \+ byte_extract_little_endian\(c, (0|0l|0ll), float\): SUCCESS
-\[main.NaN.9\] line \d+ NaN on / in byte_extract_little_endian\(c, (0|0l|0ll), float\) / myzero: SUCCESS
-\[main.NaN.10\] line \d+ NaN on \* in mynan \* mynan: SUCCESS
-\[main.NaN.11\] line \d+ NaN on \+ in mynan \+ mynan: SUCCESS
-\[main.NaN.12\] line \d+ NaN on - in mynan - mynan: SUCCESS
-\[main.NaN.13\] line \d+ NaN on / in mynan / mynan: SUCCESS
-\[main.NaN.14\] line \d+ NaN on \+ in mynan \+ \(float\)n: SUCCESS
-\[main.NaN.15\] line \d+ NaN on - in mynan - \(float\)n: SUCCESS
-\[main.NaN.16\] line \d+ NaN on \* in mynan \* \(float\)n: SUCCESS
-\[main.NaN.17\] line \d+ NaN on / in mynan / \(float\)n: SUCCESS
+\[main.NaN.6\] line \d+ NaN on \+ in myinf \+ -myinf: FAILURE
```

Comment:
Actually; this raises an interesting question.  Floating-point addition and multiplication are commutative but not associative.  If seems like normalising / sorting these would be worth while.

Reply:
I'm not sure I am able to follow: why would sorting (which this PR largely disables) be particularly beneficial for those operations? We surely can selectively re-enable sorting, but I don't think I understand the rationale just yet.

Reply:
```
float a;
float b;
if (!isnan(a) && !isnan(b)) {
  assert(a + b === b + a);
}
```

Normalising the order of commutative-but-not-associative operations means we can resolve this at simplification, rather than an expensive solver call.  It is not a massively common case but it is very cheap to implement and IIRC is a huge speed up in this case.  Note that this can be purely local as it is not associative.

Reply:
It seems we didn't actually simplify such expressions before as commutative-but-not-associative operations were exempt. I have now added 375f208 to take care of this, thank you very much for raising this!